fem/linearform.cpp

```diff
@@ -155,18 +157,32 @@ bool LinearForm::SupportsDevice()
    return true;
 }
 
-void LinearForm::Assemble(bool use_device)
+void LinearForm::SetAssemblyLevel(AssemblyLevel assembly_level)
+{
+   if (ext)
+   {
+      MFEM_ABORT("the assembly level has already been set!");
+   }
+   assembly = assembly_level;
+   switch (assembly)
+   {
+      case AssemblyLevel::LEGACY:
+         break;
+      case AssemblyLevel::FULL:
+         if (SupportsDevice()) { ext = new LinearFormExtension(this); }
+         break;
+      default:
+         mfem_error("Linearform does not support the given assembly level");
```

Comment:
```suggestion
         MFEM_ABORT("Linearform does not support the given assembly level");
```

Reply:
Addressed in https://github.com/mfem/mfem/pull/3235/commits/c60263c1fd9ecaa9a437346bbb69e172939d7fa1